src/core/coap/coap_secure.cpp

```diff
@@ -26,61 +26,63 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#define WPP_NAME "secure_coap_server.tmh"
+#define WPP_NAME "coap_secure.tmh"
 
 #ifdef OPENTHREAD_CONFIG_FILE
 #include OPENTHREAD_CONFIG_FILE
 #else
 #include <openthread-config.h>
 #endif
 
-#include "secure_coap_server.hpp"
+#include "coap_secure.hpp"
 
 #include "common/logging.hpp"
 #include "meshcop/dtls.hpp"
 #include "thread/thread_netif.hpp"
 
-#if OPENTHREAD_ENABLE_COMMISSIONER && OPENTHREAD_FTD
+#if OPENTHREAD_ENABLE_JOINER
```

Comment:
Shouldn't this be
```
#if (OPENTHREAD_ENABLE_COMMISSIONER && OPENTHREAD_FTD) || OPENTHREAD_ENABLE_JOINER
```
or 
```
#if OPENTHREAD_ENABLE_DTLS
```
based on changes in `thread_netif.hpp`?
For example in situations when only the commissioner is enabled.